api/pkg/apis/v1alpha1/managers/solution/summary-cleanup-manager.go

```
@@ -83,6 +104,47 @@ func (s *SummaryCleanupManager) Poll() []error {
 	return ret
 }
 
+// getResourceCounts is a helper function to get resource counts across all namespaces
+func (s *SummaryCleanupManager) getResourceCounts(ctx context.Context) (map[string]int, error) {
+	// Define the resource types and their metadata
+	resourceMetadata := []struct {
+		Resource string
+		Group    string
+		Version  string
+	}{
+		{"instances", "solution.symphony", "v1"},
+		{"targets", "fabric.symphony", "v1"},
+		{"solutions", "solution.symphony", "v1"},
+		{"solutioncontainers", "solution.symphony", "v1"},
+	}
+
+	// Initialize a map to store resource counts
+	resourceCounts := make(map[string]int)
+
+	// Iterate over each resource type and call the StateProvider's List method
+	for _, metadata := range resourceMetadata {
+		listRequest := states.ListRequest{
+			Metadata: map[string]interface{}{
+				"group":    metadata.Group,
+				"version":  metadata.Version,
+				"resource": metadata.Resource,
+			},
+		}
+
+		// Use the provider's List method to fetch resources
+		entities, _, err := s.StateProvider.List(ctx, listRequest)
+		if err != nil {
+			log.ErrorfCtx(ctx, "Failed to list %s: %v", metadata.Resource, err)
+			return nil, fmt.Errorf("failed to list %s: %w", metadata.Resource, err)
```

This is a report thread, seems we can make the error silently fail and log it, which is enough.

---

o'kok

---

o'kok